### ISSUE TITLE
HTTP Download bug if filename contains quote

### ISSUE THREAD

**Issue by Vinceveve** [kestra-io/kestra#5918]
### Describe the issue

When the reponse header contains quotes in filename, download fails

HTTP output
```
Content-Disposition: attachment; filename="mamp-c13.gtfs.zip"
```
Error :
```
Illegal character in path at index 110: /iv/download-static-feeds/executions/13pqIRmmJbsMbw1s100AUe/tasks/download-static-feed/4bFHdvmXkekG1XSl2NrSnR/"mamp-c13.gtfs.zip"
```
Task :
```- id: download_static_feed
  type: io.kestra.plugin.core.http.Download
  options: 
    followRedirects: true
  uri: "https://www.data.gouv.fr/fr/datasets/r/10e025a1-9048-4c19-925a-466b2a79232f"
  ```

### Environment

- Kestra Version: 0.19.10

**Comment by MilosPaunovic** [comment]
Checked this one out, seems to be the problem on the BE side, with character encoding, as UI is just passing the value.

**Comment by benjoEK1337** [comment]
Hey, this seemed interesting to me, so did a investigation.

I played with unit tests, it seems like this [PR](https://github.com/kestra-io/kestra/pull/5593) broke the working functionality. 

I have run the unit test DownloadTest on the latest develop code and it was failing complaining: 

`Illegal char <"> at index 152: C:\tmp\unittest\/io/kestra/plugin/core/http/downloadtest/contentdisposition/executions/3xk3HOCaCO0ZAXBpphNTM3/tasks/downloadtest/1MTPWX3gWbuDbiKQepPDSR/"filename.jpg"
java.nio.file.InvalidPathException: Illegal char <"> at index 152: C:\tmp\unittest\/io/kestra/plugin/core/http/downloadtest/contentdisposition/executions/3xk3HOCaCO0ZAXBpphNTM3/tasks/downloadtest/1MTPWX3gWbuDbiKQepPDSR/"filename.jpg"
`

I then reverted this line (which is the latest develop)

![image](https://github.com/user-attachments/assets/cffcf74e-a49c-4f15-b22d-5b536114833e)

to old code before the change:

![image](https://github.com/user-attachments/assets/41c243e0-9b40-4f37-91b2-cfb6dde1c7b0)

and the test is passing.

From my understanding this new change is basically saying include the " characters. Maybe there is a hidden reason why this is done, but I think this line could be reverted and it would work correctly (the line related to adding the encoding could be kept)

**Comment by loicmathieu** [comment]
@Skraye can you give insight on why you did that change?

**Comment by Skraye** [comment]
This removes first and last letter, that's why it worked with your specific usecases, but with a normal filename, result is not good : 
![image](https://github.com/user-attachments/assets/10e4c700-6f6c-42aa-987b-a8ad3740521c)

**Comment by benjoEK1337** [comment]
@Skraye I didn't understand what you meant 😢 Or maybe you didn't reply to my comment 😄 

So.. to rephrase, is it safe to revert to the old code the part related to creating the substring/filename? (look at my comment above)

**Comment by Skraye** [comment]
This comment explained why this code won't be reversed.

Tldr; you filename is wrong the code is correct

**Comment by Ben8t** [comment]
@benjoEK1337 what @Skraye meant in this [comment](https://github.com/kestra-io/kestra/issues/5918#issuecomment-2482302197) is that your provided fix isn't working: with "normal" filename, it will replace first and last character (see the screenshot with `hello.txt` replace by `ello.tx`)

**Comment by Vinceveve** [comment]
Maybe it could be a missing feature : force filename ?
This way I can avoid the (maybe) bug by setting a desired filename for the output.
What do you think ?

**Comment by Ben8t** [comment]
@Skraye wdyt?

`"` is indeed a special character but it could fixed by addind escape char or something. Especially knowning that it was working before this PR (as per [this](https://github.com/kestra-io/kestra/issues/5918#issuecomment-2479000067) comment)

**Comment by loicmathieu** [comment]
Fixed by https://github.com/kestra-io/kestra/pull/6291/files which is already included in the latest 0.20.3 release.
